frontend/packages/dev-console/src/components/topology/TopologyView.tsx

```diff
@@ -0,0 +1,394 @@
+import TopologyResourcePanel from './TopologyResourcePanel';
+import {
+  TYPE_EVENT_PUB_SUB_LINK,
+  TYPE_EVENT_SOURCE_LINK,
+  TYPE_REVISION_TRAFFIC,
+} from '@console/knative-plugin/src/topology/const';
+import KnativeResourceOverviewPage from '@console/knative-plugin/src/components/overview/KnativeResourceOverviewPage';
+import KnativeTopologyEdgePanel from '@console/knative-plugin/src/components/overview/KnativeTopologyEdgePanel';
+import ConnectedTopologyEdgePanel from './TopologyEdgePanel';
+import TopologySideBar from './TopologySideBar';
+
+export const FILTER_ACTIVE_CLASS = 'odc-m-filter-active';
+const TOPOLOGY_GRAPH_ID = 'odc-topology-graph';
+
+interface StateProps {
+  application: string;
+  eventSourceEnabled: boolean;
+}
+
+interface DispatchProps {
+  onSelectTab?: (name: string) => void;
+  onFiltersChange: (filters: DisplayFilters) => void;
+  onSupportedFiltersChange: (supportedFilterIds: string[]) => void;
+}
+
+interface TopologyViewProps {
+  model: Model;
+  namespace: string;
+  showGraphView: boolean;
+}
+
+type ComponentProps = TopologyViewProps & StateProps & DispatchProps;
+
+const graphModel: Model = {
+  graph: {
+    id: TOPOLOGY_GRAPH_ID,
+    type: 'graph',
+    layout: COLA_LAYOUT,
+    layers: [BOTTOM_LAYER, GROUPS_LAYER, 'groups2', DEFAULT_LAYER, TOP_LAYER],
+  },
+};
+
+export const TopologyView: React.FC<ComponentProps> = ({
+  model,
+  namespace,
+  showGraphView,
+  eventSourceEnabled,
+  application,
+  onSelectTab,
+  onFiltersChange,
+  onSupportedFiltersChange,
+}) => {
+  const [filteredModel, setFilteredModel] = React.useState<Model>();
+  const [graphData, setGraphData] = React.useState<GraphData>();
+  const [storedSelectedIds, setSelectedIds] = React.useState<string[]>([]);
+  const selectedIds = useDeepCompareMemoize(storedSelectedIds);
+  const createResourceAccess: string[] = useAddToProjectAccess(namespace);
+  const filters = useDisplayFilters();
+  const appliedFilters = useAppliedDisplayFilters();
+  const [displayFilterers, setDisplayFilterers] = React.useState<TopologyApplyDisplayOptions[]>(
+    null,
+  );
+  const displayFilterExtensions = useExtensions<TopologyDisplayFilters>(isTopologyDisplayFilter);
+  const createConnectorExtensions = useExtensions<TopologyCreateConnector>(
+    isTopologyCreateConnector,
+  );
+  const [createConnectors, setCreateConnectors] = React.useState<CreateConnectionGetter[]>(null);
+  const searchParams = window.location.search;
+  const [filtersLoaded, setFiltersLoaded] = React.useState<boolean>(false);
+  const visualizationRef = React.useRef<Visualization>();
+  const queryParams = useQueryParams();
+
+  const createConnectorPromises = React.useMemo(
+    () => createConnectorExtensions.map((creator) => creator.properties.getCreateConnector()),
+    [createConnectorExtensions],
+  );
+
+  React.useEffect(() => {
+    if (createConnectorPromises) {
+      if (createConnectorPromises.length === 0) {
```

Comment:
You can skip this check because `Promise.all([])` will also run the then case with an empty result, or you should return after `setCreateConnectors([]);` The current code will call `setCreateConnectors` twice.